lib/ukstore/include/uk/store.h

```diff
@@ -29,3 +29,345 @@
+	_LIB_UK_STORE_ENTRY_TYPE___s8,
+	_LIB_UK_STORE_ENTRY_TYPE___u8,
+	_LIB_UK_STORE_ENTRY_TYPE___s16,
+	_LIB_UK_STORE_ENTRY_TYPE___u16,
+	_LIB_UK_STORE_ENTRY_TYPE___s32,
+	_LIB_UK_STORE_ENTRY_TYPE___u32,
+	_LIB_UK_STORE_ENTRY_TYPE___s64,
+	_LIB_UK_STORE_ENTRY_TYPE___u64,
+	_LIB_UK_STORE_ENTRY_TYPE___uptr,
+	_LIB_UK_STORE_ENTRY_TYPE___charp
+};
+
+/* Transforms a simple data type to a ukstore specific one */
+#define UK_STORE_ENTRY_TYPE(simple_type)	\
+	(_LIB_UK_STORE_ENTRY_TYPE___ ## simple_type)
+
+/* Getter definitions */
+typedef int (*uk_store_get_s8_func_t)(void *, __s8 *);
+typedef int (*uk_store_get_u8_func_t)(void *, __u8 *);
+typedef int (*uk_store_get_s16_func_t)(void *, __s16 *);
+typedef int (*uk_store_get_u16_func_t)(void *, __u16 *);
+typedef int (*uk_store_get_s32_func_t)(void *, __s32 *);
+typedef int (*uk_store_get_u32_func_t)(void *, __u32 *);
+typedef int (*uk_store_get_s64_func_t)(void *, __s64 *);
+typedef int (*uk_store_get_u64_func_t)(void *, __u64 *);
+typedef int (*uk_store_get_uptr_func_t)(void *, __uptr *);
+typedef int (*uk_store_get_charp_func_t)(void *, char **);
+typedef int (*uk_store_get_ncharp_func_t)(void *, char **, size_t);
+
+/* Setter definitions */
+typedef int (*uk_store_set_s8_func_t)(void *, __s8);
+typedef int (*uk_store_set_u8_func_t)(void *, __u8);
+typedef int (*uk_store_set_s16_func_t)(void *, __s16);
+typedef int (*uk_store_set_u16_func_t)(void *, __u16);
+typedef int (*uk_store_set_s32_func_t)(void *, __s32);
+typedef int (*uk_store_set_u32_func_t)(void *, __u32);
+typedef int (*uk_store_set_s64_func_t)(void *, __s64);
+typedef int (*uk_store_set_u64_func_t)(void *, __u64);
+typedef int (*uk_store_set_uptr_func_t)(void *, __uptr);
+typedef int (*uk_store_set_charp_func_t)(void *, const char *);
+
+/* Cleanup definition */
+typedef void (*uk_store_entry_freed_func_t)(void *);
+
+struct uk_store_entry {
+	/* Function getter pointer */
+	union {
+		uk_store_get_s8_func_t     s8;
+		uk_store_get_u8_func_t     u8;
+		uk_store_get_s16_func_t    s16;
+		uk_store_get_u16_func_t    u16;
+		uk_store_get_s32_func_t    s32;
+		uk_store_get_u32_func_t    u32;
+		uk_store_get_s64_func_t    s64;
+		uk_store_get_u64_func_t    u64;
+		uk_store_get_uptr_func_t   uptr;
+		uk_store_get_charp_func_t  charp;
+	} get;
+
+	/* Function setter pointer */
+	union {
+		uk_store_set_s8_func_t    s8;
+		uk_store_set_u8_func_t    u8;
+		uk_store_set_s16_func_t   s16;
+		uk_store_set_u16_func_t   u16;
+		uk_store_set_s32_func_t   s32;
+		uk_store_set_u32_func_t   u32;
+		uk_store_set_s64_func_t   s64;
+		uk_store_set_u64_func_t   u64;
+		uk_store_set_uptr_func_t  uptr;
+		uk_store_set_charp_func_t charp;
+	} set;
+
+	/* The entry name */
+	char *name;
+
+	/* Entry flags */
+	int flags;
+
+	/* Entry getter/setter type */
+	enum uk_store_entry_type type;
+
+	/* A cookie for extra storage */
+	void *cookie;
+} __align8;
+
+/* Flags if an entry is static or dynamic */
+#define UK_STORE_ENTRY_FLAG_STATIC	(1 << 0)
```

Comment:
in order to avoid duplication, put common defintions (when library is on or off) before the guard checking if the library is part of the built: line 48

Reply:
I think there is nothing that is common between the two if/else cases. I'll just move the `#endif` lower.